llm_bioinformatics_research/src/Testing/ResetPassword.test.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import ResetPassword from '../Components/ResetPassword/ResetPassword';
+import { BrowserRouter } from 'react-router-dom';
+
+global.fetch = jest.fn();
+const setup = () => render(
+  <BrowserRouter>
+    <ResetPassword />
+  </BrowserRouter>
+);
+
+describe("ResetPassword Component", () => {
+  beforeEach(() => {
+    fetch.mockClear();
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.runOnlyPendingTimers();
+    jest.useRealTimers();
+  });
+
+  it("submits form with valid password and displays success message, then redirects", async () => {
+    fetch.mockResolvedValueOnce({
+      status: 200,
+      json: async () => ({ message: "Password reset successful!" }),
+    });
+
+    setup();
+    const newPasswordInput = screen.getByLabelText("New Password");
+    const confirmPasswordInput = screen.getByLabelText("Confirm Password");
+    const resetButton = screen.getByRole("button", { name: /reset password/i });
+
+    userEvent.type(newPasswordInput, "ValidPass123!");
+    userEvent.type(confirmPasswordInput, "ValidPass123!");
+    fireEvent.click(resetButton);
+
+    await waitFor(() => expect(screen.getByText("Password reset successful!")).toBeInTheDocument());
+    jest.advanceTimersByTime(3000);
+    expect(window.location.pathname).toBe("/login");
+  });
+});
+
+describe("ResetPassword Component - Password Validation", () => {
+  beforeEach(() => {
+    setup();
+  });
+
+  it("shows error when password is less than 8 characters", () => {
+    const newPasswordInput = screen.getByLabelText("New Password");
+    const confirmPasswordInput = screen.getByLabelText("Confirm Password");
+    const resetButton = screen.getByRole("button", { name: /reset password/i });
+
+    userEvent.type(newPasswordInput, "Short1!");
+    userEvent.type(confirmPasswordInput, "Short1!");
+    fireEvent.click(resetButton);
+
+    expect(screen.getByText("Password must be at least 8 characters long.")).toBeInTheDocument();
+  });
+
```

Comment:
As an addition to the comment I made earlier, we should a test to make sure that the new password is not the same as the existing password in the database. If it is, then the user should be prompted to modify the new password and we should add a test to check for this.